### PR TITLE
Docs updates + two additional unit tests

### PR DESCRIPTION
**Pull request by vblagoje**
- fixes https://github.com/deepset-ai/haystack/issues/7127

### CODE REVIEW

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/generator.py [hunk 1]
```diff
@@ -193,21 +222,37 @@ def invoke(self, *args, **kwargs):
 
         return responses
 
-    @component.output_types(replies=List[str], metadata=List[Dict[str, Any]])
+    @component.output_types(replies=List[str])
```
Comment (vblagoje) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/generator.py:225]: @davidsbatista we missed this one during previous review, there is not metadata output

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py [hunk 2]
```diff
@@ -166,21 +175,37 @@ def invoke(self, *args, **kwargs):
 
         return responses
 
-    @component.output_types(replies=List[str], metadata=List[Dict[str, Any]])
+    @component.output_types(replies=List[ChatMessage])
```
Comment (vblagoje) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py:178]: And here the output type wasn't  correct + there is no metadata

File: integrations/amazon_bedrock/tests/test_chat_generator.py [hunk 3]
```diff
@@ -173,6 +173,18 @@ def test_prepare_body_with_custom_inference_params(self) -> None:
...
         assert body == expected_body
 
+    @pytest.mark.integration
+    def test_get_responses(self) -> None:
+        adapter = AnthropicClaudeChatAdapter(generation_kwargs={})
+        response_body = {"completion": "This is a single response."}
+        expected_response = "This is a single response."
+        response_message = adapter.get_responses(response_body)
+        # assert that the type of each item in the list is a ChatMessage
+        for message in response_message:
```
Comment (vblagoje) [inline integrations/amazon_bedrock/tests/test_chat_generator.py:183]: @davidsbatista added tests to confirm the types

**Review by vblagoje — COMMENTED** [review]
@davidsbatista I made two small changes - please run the generators again to confirm that they work

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/generator.py [hunk 4]
```diff
@@ -33,16 +33,18 @@
...
+    For example, to use the Anthropic Claude model, simply initialize the `AmazonBedrockGenerator` with the
+    'anthropic.claude-v2' model name. Provide AWS credentials either via local AWS profile or directly via
+    `aws_access_key_id`, `aws_secret_access_key`, `aws_session_token`, and `aws_region_name` parameters.
 
     Usage example:
     ```python
-    from amazon_bedrock_haystack.generators.amazon_bedrock import AmazonBedrockGenerator
+    from haystack_integrations.components.generators.amazon_bedrock import AmazonBedrockGenerator
 
     generator = AmazonBedrockGenerator(
```
Comment (davidsbatista) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/generator.py:46]: The AWS creds are no longer strings, they are either `Secret` or env var - you can omit them.

```suggestion
generator = AmazonBedrockGenerator(
        model="anthropic.claude-v2",
        max_length=99,
)
```

File: integrations/amazon_bedrock/tests/test_chat_generator.py (hunk 3 above)
Comment (davidsbatista) [reply integrations/amazon_bedrock/tests/test_chat_generator.py:183]: 👍🏽

**Comment by vblagoje** [comment]
Ok @davidsbatista please double check the latest changes, I had issues with linter

**Review by davidsbatista — APPROVED** [review]
LGTM

**Pull request merged**